### PR TITLE
bpo-45886: Allow overriding freeze command for cross compiling (GH-29735)

### PR DESCRIPTION
**Pull request by tiran**
```
../../configure -C --host=aarch64-linux-gnu-gcc --build=x86_64-pc-linux-gnu
make FREEZE_MODULE=../x86_64/Program/_freeze_module
```

Signed-off-by: Christian Heimes <christian@python.org>

https://bugs.python.org/issue45886

### CODE REVIEW

**Comment by tiran** [comment]
```
$ rm Python/frozen_modules/*.h
$ make regen-deepfreeze FREEZE_MODULE=builddep/origin/Programs/_freeze_module
builddep/origin/Programs/_freeze_module importlib._bootstrap ./Lib/importlib/_bootstrap.py Python/frozen_modules/importlib._bootstrap.h
builddep/origin/Programs/_freeze_module importlib._bootstrap_external ./Lib/importlib/_bootstrap_external.py Python/frozen_modules/importlib._bootstrap_external.h
builddep/origin/Programs/_freeze_module zipimport ./Lib/zipimport.py Python/frozen_modules/zipimport.h
...
```

File: Makefile.pre.in [hunk 1]
```diff
@@ -1061,69 +1064,69 @@ Programs/_freeze_module: Programs/_freeze_module.o $(LIBRARY_OBJS_OMIT_FROZEN)
...
 
-Python/frozen_modules/frozen_only.h: Programs/_freeze_module Tools/freeze/flag.py
-	Programs/_freeze_module frozen_only $(srcdir)/Tools/freeze/flag.py Python/frozen_modules/frozen_only.h
+Python/frozen_modules/frozen_only.h: $(FREEZE_MODULE) Tools/freeze/flag.py
+	$(FREEZE_MODULE) frozen_only $(srcdir)/Tools/freeze/flag.py Python/frozen_modules/frozen_only.h
 
 # END: freezing modules
 
-Tools/scripts/freeze_modules.py: Programs/_freeze_module
+Tools/scripts/freeze_modules.py: $(FREEZE_MODULE)
```
Comment (gvanrossum) [inline Makefile.pre.in:1129]: I guess when Make sees a reference to something outside the tree it's just going to take it at face value, assuming it exists at all? What's the failure mode if that file doesn't exist?

**Review by gvanrossum — APPROVED** [review]
LGTM.

File: Makefile.pre.in (hunk 1 above)
Comment (tiran) [reply Makefile.pre.in:1129]: Our Makefile does not specify a build rule for the target ``$(FREEZE_MODULE)``. There is only a build rule for ``Programs/_freeze_module``. If ``FREEZE_MODULE`` points to a different, non-existing file then ``make`` complains and stops:

```
$ make regen-deepfreeze  FREEZE_MODULE=builddep/origin/Programs/invalid
make: *** No rule to make target 'builddep/origin/Programs/invalid', needed by 'Python/frozen_modules/importlib._bootstrap.h'.  Stop.
```

**Pull request merged**